specfile/changelog.py

```diff
@@ -361,3 +366,49 @@ def get_raw_section_data(self) -> List[str]:
             result.extend(entry.content)
             result.extend(entry._following_lines)
         return result
+
+
+def _getent_name() -> str:
+    username = getpass.getuser()
+    pwd_struct = pwd.getpwnam(username)
+    # Use the plain username if the name field is empty
+    return pwd_struct.pw_gecos or username
+
+
+def guess_packager() -> str:
+    """
+    Guess the name and email of a packager to use for changelog entries.
+    This uses similar logic to rpmdev-packager.
+    The following places are searched for this value (in this order):
+        - $RPM_PACKAGER envvar
+        - %packager macro
+        - git config
+        - Unix username
+
+    Returns:
+        A string to use for the changelog entry author.
+        If nothing was detected, an empty string is returned.
+    """
+    if "RPM_PACKAGER" in os.environ:
+        return os.environ["RPM_PACKAGER"]
+
+    packager = rpm.expandMacro("%packager")
```

Comment:
There is [`Macros.expand()`](https://github.com/packit/specfile/blob/adfd72d9d1394faaa2e2cafc4a3d244a82c5131f/specfile/macros.py#L134-L151) that wraps `rpm.expandMacro()` and ensures any errors written by RPM on stderr are captured in an exception, it would be better to use that.

Reply:
https://github.com/packit/specfile/pull/220/commits/dbbc53a79a46f26251a36de275a42a927b5f472a